shared/src/commonMain/kotlin/org/hyperskill/app/progresses/domain/interactor/ProgressesInteractor.kt

```diff
@@ -8,21 +8,21 @@ import org.hyperskill.app.track.domain.model.TrackProgress
 class ProgressesInteractor(
     private val progressesRepository: ProgressesRepository
 ) {
-    suspend fun getTracksProgresses(tracksIds: List<Long>): Result<List<TrackProgress>> =
-        progressesRepository.getTracksProgresses(tracksIds)
+    suspend fun getTracksProgresses(tracksIds: List<Long>, forceLoadFromRemote: Boolean): Result<List<TrackProgress>> =
+        progressesRepository.getTracksProgresses(tracksIds, forceLoadFromRemote)
 
-    suspend fun getTrackProgress(trackId: Long): Result<TrackProgress?> =
-        progressesRepository.getTrackProgress(trackId)
+    suspend fun getTrackProgress(trackId: Long, forceLoadFromRemote: Boolean): Result<TrackProgress?> =
+        progressesRepository.getTrackProgress(trackId, forceLoadFromRemote)
 
     suspend fun getTopicsProgresses(topicsIds: List<Long>): Result<List<TopicProgress>> =
         progressesRepository.getTopicsProgresses(topicsIds)
 
     suspend fun getTopicProgress(topicId: Long): Result<TopicProgress> =
         progressesRepository.getTopicProgress(topicId)
 
-    suspend fun getProjectsProgresses(projectsIds: List<Long>): Result<List<ProjectProgress>> =
-        progressesRepository.getProjectsProgresses(projectsIds)
+    suspend fun getProjectsProgresses(projectsIds: List<Long>, forceLoadFromRemote: Boolean): Result<List<ProjectProgress>> =
+        progressesRepository.getProjectsProgresses(projectsIds, forceLoadFromRemote)
 
-    suspend fun getProjectProgress(projectId: Long): Result<ProjectProgress> =
-        progressesRepository.getProjectProgress(projectId)
+    suspend fun getProjectProgress(projectId: Long, forceLoadFromRemote: Boolean): Result<ProjectProgress> =
+        progressesRepository.getProjectProgress(projectId, forceLoadFromRemote)
```

Comment:
if we don't have any business logic and just delegate in interactor, why can't we use progressesRepository directly?

Reply:
I agree. I don't know the reason we have `ProgressInteractor`. The major of its methods is not used.